### PR TITLE
DO NOT MERGE - POC of nginx centos container

### PR DESCRIPTION
**Pull request by tompizmor**
**Description of the change**

This PR shows how would look like the repository for nginx providing centos and debian images. The base image for the centos option is based on James' work in the `centos-demo` repository. We should start productizing our centos base image at some point.

### CODE REVIEW

File: 1.12/centos/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+FROM tompizmor/centos:7
```
Comment (sebgoa) [inline 1.12/centos/Dockerfile]: we should use the official image and install our things transparently.

Comment (sameersbn) [reply 1.12/centos/Dockerfile]: @sebgoa It appears `tompizmor/centos:7` contains the nami package manager and other supporting scripts.

File: 1.12/centos/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+LABEL maintainer "Bitnami <containers@bitnami.com>"
+
+ENV BITNAMI_PKG_CHMOD="-R g+rwX" \
+    BITNAMI_PKG_EXTRA_DIRS="/bitnami/nginx/conf" \
+    HOME="/"
+
+# Install required system packages and dependencies
+RUN yum install libc pcre openlibssl zlib
```
Comment (sameersbn) [inline 1.12/centos/Dockerfile]: Looks like it worked without it, but shouldn't this have been `yum install -y` ?

File: 1.12/debian/rootfs/app-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+. /opt/bitnami/base/functions
+. /opt/bitnami/base/helpers
+
+print_welcome_page
+check_for_updates &
```
Comment (sameersbn) [inline 1.12/debian/rootfs/app-entrypoint.sh]: remove this and associated logic, see https://github.com/bitnami/minideb-extras/pull/133

File: circle.yml [hunk 4]
```diff
@@ -35,22 +36,30 @@ jobs:
...
       - save_cache:
           key: cache-{{ .Branch }}-{{ epoch }}
           paths:
             - /cache/layers.tar
       - deploy:
           name: Publish docker image
           command: |
+            export CIRCLE_TAG=1.12.1-rTest3
+            echo ${CIRCLE_TAG}
+            echo ${CIRCLE_BRANCH}
```
Comment (sameersbn) [inline circle.yml]: Looks like temporary/test bits that need to go

File: circle.yml [hunk 5]
```diff
@@ -6,19 +6,20 @@ jobs:
...
           RELEASE_SERIES_LIST: "1.12"
+          SUPPORTED_BASE_IMAGES: "debian,centos"
           LATEST_STABLE: "1.12"
           IMAGE_NAME: nginx
-          CHART_NAME: nginx
-          CHART_REPO: https://github.com/bitnami/charts
-          DOCKER_PROJECT: bitnami
-          QUAY_PROJECT: bitnami
-          GCLOUD_PROJECT: bitnami-containers
+          DOCKER_PROJECT: tompizmor
```
Comment (sameersbn) [inline circle.yml]: This change will be reverted before merge, correct?

File: circle.yml (hunk 4 above)
Comment (tompizmor) [reply circle.yml]: right

File: circle.yml (hunk 5 above)
Comment (tompizmor) [reply circle.yml]: yep

File: 1.12/centos/Dockerfile (hunk 2 above)
Comment (tompizmor) [reply 1.12/centos/Dockerfile]: yes, I think the same

File: 1.12/debian/rootfs/app-entrypoint.sh (hunk 3 above)
Comment (tompizmor) [reply 1.12/debian/rootfs/app-entrypoint.sh]: done

File: 1.12/centos/Dockerfile (hunk 1 above)
Comment (tompizmor) [reply 1.12/centos/Dockerfile]: yes, apart from installing nami and tini, it copies the following tree directory:

```
rootfs/
├── entrypoint.sh
├── opt
│   └── bitnami
│       ├── base
│       │   ├── functions
│       │   └── helpers
│       └── updates-ca-cert.pem
└── usr
    └── local
        └── bin
            ├── bitnami-pkg
            └── print-help
```

**Pull request closed without merging**